includes/api/class-wc-rest-orders-controller.php

```diff
@@ -1656,4 +1678,64 @@ public function get_collection_params() {
 
 		return $params;
 	}
+
+	/**
+	 * Does a permissions check for the proper requested blog
+	 *
+	 * @param WP_REST_Request $request
+	 *
+	 * @return bool $permission
+	 */
+	public function network_orders_permissions_check( $request ) {
+		$blog_id = $request->get_param( 'blog_id' );
+		$blog_id = $blog_id ?: get_current_blog_id();
+
+		switch_to_blog( $blog_id );
+
+		$permission = $this->get_items_permissions_check( $request );
+
+		restore_current_blog();
+
+		return $permission;
+	}
+
+	/**
+	 * @param WP_REST_Request $request
+	 *
+	 * @return mixed|WP_REST_Response
+	 */
+	public function network_orders( $request ) {
+		$blog_id = $request->get_param( 'blog_id' );
+		$blog_id = $blog_id ?: get_current_blog_id();
+
+		switch_to_blog( $blog_id );
+
+		add_filter( 'woocommerce_rest_orders_prepare_object_query', array( $this, 'network_orders_filter_args' ) );
+		$items = $this->get_items( $request );
+		remove_filter( 'woocommerce_rest_orders_prepare_object_query', array( $this, 'network_orders_filter_args' ) );
+
+		foreach( $items->data as &$current_order ) {
+			$order = wc_get_order( $current_order['id'] );
+
+			$current_order['blog'] = get_blog_details( get_current_blog_id() );
```

Comment:
This items included here that need to belong the schema and that's why will be better have a new class just for it.